packages/comms/src/ecl/logicalFile.ts

```diff
@@ -0,0 +1,105 @@
+    get Dir(): string { return this.get("Dir"); }
+    get PathMask(): string { return this.get("PathMask"); }
+    get Filesize(): string { return this.get("Filesize"); }
+    get FileSizeInt64(): number { return this.get("FileSizeInt64"); }
+    get RecordSize(): string { return this.get("RecordSize"); }
+    get RecordCount(): string { return this.get("RecordCount"); }
+    get RecordSizeInt64(): number { return this.get("RecordSizeInt64"); }
+    get RecordCountInt64(): number { return this.get("RecordCountInt64"); }
+    get Wuid(): string { return this.get("Wuid"); }
+    get Owner(): string { return this.get("Owner"); }
+    get JobName(): string { return this.get("JobName"); }
+    get Persistent(): string { return this.get("Persistent"); }
+    get Format(): string { return this.get("Format"); }
+    get MaxRecordSize(): string { return this.get("MaxRecordSize"); }
+    get CsvSeparate(): string { return this.get("CsvSeparate"); }
+    get CsvQuote(): string { return this.get("CsvQuote"); }
+    get CsvTerminate(): string { return this.get("CsvTerminate"); }
+    get CsvEscape(): string { return this.get("CsvEscape"); }
+    get Modified(): string { return this.get("Modified"); }
+    get Ecl(): string { return this.get("Ecl"); }
+    get Stat(): WsDfu.Stat { return this.get("Stat"); }
+    get DFUFilePartsOnClusters(): WsDfu.DFUFilePartsOnClusters { return this.get("DFUFilePartsOnClusters"); }
+    get isSuperfile(): boolean { return this.get("isSuperfile"); }
+    get ShowFileContent(): boolean { return this.get("ShowFileContent"); }
+    get subfiles(): WsDfu.Subfiles { return this.get("subfiles"); }
+    get Superfiles(): WsDfu.Superfiles { return this.get("Superfiles"); }
+    get ProtectList(): WsDfu.ProtectList { return this.get("ProtectList"); }
+    get FromRoxieCluster(): boolean { return this.get("FromRoxieCluster"); }
+    get Graphs(): WsDfu.Graphs { return this.get("Graphs"); }
+    get UserPermission(): string { return this.get("UserPermission"); }
+    get ContentType(): string { return this.get("ContentType"); }
+    get CompressedFileSize(): number { return this.get("CompressedFileSize"); }
+    get PercentCompressed(): string { return this.get("PercentCompressed"); }
+    get IsCompressed(): boolean { return this.get("IsCompressed"); }
+    get BrowseData(): boolean { return this.get("BrowseData"); }
+    get jsonInfo(): string { return this.get("jsonInfo"); }
+    get binInfo(): string { return this.get("binInfo"); }
+    get PackageID(): string { return this.get("PackageID"); }
+    get Partition(): WsDfu.Partition { return this.get("Partition"); }
+    get Blooms(): WsDfu.Blooms { return this.get("Blooms"); }
+    get ExpireDays(): number { return this.get("ExpireDays"); }
+    get KeyType(): string { return this.get("KeyType"); }
+
+    get properties(): FileDetailEx { return this.get(); }
+
+    static attach(optsConnection: IOptions | IConnection | DFUService, Cluster: string, Name: string): LogicalFile {
```

Comment:
What is this attach method doing? Attaching one logical file to another?

Reply:
Note that method is "static singleton factory" and that the constructor is "private" - so the pattern is:
```
const logicalFileRef = LogicalFile.attach(baseUrl, ID);
```

And the following is impossible:
```
const logicalFileRef = new LogicalFile(baseUrl, ID);
```

The idea is to ensure that there can only ever be 1 instance of the LogicalFile class per unique logical file.

Reply:
And then you'd use it like this?
```javascript
const baseUrl = "https://play.hpccsystems.com:18010";
const ID = ".::kaggle_county_crimes.csv";
const logicalFileRef = LogicalFile.attach(baseUrl, ID);
const infoObj = logicalFileRef.fetchInfo();
console.log(infoObj);
```

Reply:
Nearly:
```
const logicalFileRef = LogicalFile.attach({baseUrl:  "https://play.hpccsystems.com:18010"}, ".::kaggle_county_crimes.csv");
...
```